src/AuthenticationServiceInterface.php

```diff
@@ -17,12 +17,12 @@
 namespace Authentication;
 
 use Authentication\Authenticator\AuthenticatorInterface;
-use Authentication\Authenticator\PersistenceInterface;
+use Authentication\Authenticator\ImpersonationInterface;
 use Authentication\Authenticator\ResultInterface;
 use Authentication\Identifier\IdentifierInterface;
 use Psr\Http\Message\ServerRequestInterface;
 
-interface AuthenticationServiceInterface extends PersistenceInterface
+interface AuthenticationServiceInterface extends ImpersonationInterface
```

Comment:
If `AuthenticationServiceInterface` extends `ImpersonationInterface` then perhaps it should be under the namespace `Authentication` instead of `Authentication\Authenticator`.

Also changing the interface is not backwards compatible and would require a major version bump. To avoid that this extension would have to be keep and have `AuthenticationService` directly implement `ImpersonationInterface` (which is already done currently).

Reply:
:+1: to adding a new interface instead of modifying the existing one.